backend/src/api/streak.ts

```diff
@@ -0,0 +1,37 @@
+import { Request, Response } from "express";
+import { calculateStreak } from "../models/streakCalculation.js";
+import { verifyAndDecodeToken } from "./auth/sharedAuth.js";
+
+export async function fetchStreakHandler(req: Request, res: Response) {
+  const userData = verifyAndDecodeToken(req.cookies.token);
+
+  if (!userData) {
+    res.status(401).json({ error: "Unauthorized" });
+    return;
+  }
+
+  try {
+    const streak = await calculateStreak(userData.id);
+    res.status(200).json({ streak });
+  } catch (error) {
+    console.error("Error fetching streak:", error);
+    res.status(500).json({ error: "Internal Server Error" });
+  }
+}
+
+export async function updateStreakHandler(req: Request, res: Response) {
+    const userData = verifyAndDecodeToken(req.cookies.token);
+  
+    if (!userData) {
+      res.status(401).json({ error: "Unauthorized" });
+      return;
+    }
+  
+    try {
+      const streak = await calculateStreak(userData.id);
+      res.status(200).json({ streak });
+    } catch (error) {
+      console.error("Error updating streak:", error);
+      res.status(500).json({ error: "Internal Server Error" });
+    }
```

Comment:
What is the difference between these two functions?

Reply:
There is really no difference other than the error message. I do all of the calculation stuff in the streakCalculation.

Reply:
Oh i see, every time we get the streak we also update it, that makes sense.